File: examples/multiplier.cxx
#include <iostream>
#include <mpi.h>

#include <inC2.h>

int main(int argc, char** argv)
{
   InC2 *parent = new InC2();

   int val = atoi(argv[1]);
   int retVal = 0;
   std::cout << "Multiplier will multiply all input by " << val << " and then return the result when requested." << std::endl;

   while(1)
   {
      Message msg = parent->receiveParentMessage();
      if (!msg.getCommand().compare("STOP"))
      {
         std::cout << "Multiplier received STOP command. Leaving now" << std::endl;
         break;
      }
      else if (!msg.getCommand().compare("INPUT"))
      {
         int payload = stoi(msg.getPayload()); // we should be passing a number
         retVal = val * payload;
      }
      else if (!msg.getCommand().compare("REPORT"))
      {
         parent->sendParentMessage(std::to_string(retVal));
      }
   }

   delete parent;
   return 0;
}

File: examples/multiplier.py
import InC2

import sys

parent = InC2.InC2()

val = int(sys.argv[1])
retVal = 0
print("Multiplier will multiply all input by " + str(val) + " and then return the result when requested.")

while True:
   msg = parent.receiveParentMessage();
   if msg.getCommand() == "STOP":
      print("Multiplier received STOP command. Leaving now")
      break
   elif msg.getCommand() == "INPUT":
      payload = int(msg.getPayload()) # we should be passing a number
      retVal = val * payload
   elif msg.getCommand() == "REPORT":
      parent.sendParentMessage(str(retVal))

del parent

